### DISCUSSION TITLE
Loading properties from environment variables?

### DISCUSSION THREAD

**Discussion by fhennig** [eclipse-edc/Connector#3065] [Q&A] [answered]
Is it possible to load properties from environment variables? For example [Trino](https://trino.io/docs/current/security/secrets.html) supports this.

It would be useful to pass in secrets (like S3 keys) this way - or is there a different mechanism that should be used in this case?

**Comment by ndr-brt** [comment, marked as answer]
The environment variables are in fact read as configuration settings our of the box, converting the format
So if you pass to the environment variable `WEB_HTTP_PORT` it gets read as the setting `web.http.port`.

Any other need as environment variable replacement could be implemented in a custom extension.

**Reply by fhennig** [reply]
Ah good to know! Thanks!
